src/Explorer/Explorer.tsx

```diff
@@ -1389,9 +1390,20 @@ export default class Explorer {
     if (userContext.apiType !== "SQL" || !userContext.subscriptionId) {
       return;
     }
-    const copilotEnabled = await getCopilotEnabled();
-    useQueryCopilot.getState().setCopilotEnabled(copilotEnabled);
-    useQueryCopilot.getState().setCopilotUserDBEnabled(copilotEnabled);
+    const copilotEnabledPromise = getCopilotEnabled();
+    const copilotUserDBEnabledPromise = isCopilotFeatureRegistered(userContext.subscriptionId);
+    const [copilotEnabled, copilotUserDBEnabled] = await Promise.all([
+      copilotEnabledPromise,
+      copilotUserDBEnabledPromise,
+    ]);
+    const copilotSampleDBEnabled = LocalStorageUtility.hasItem(StorageKey.CopilotSampleDBEnabled)
+      ? LocalStorageUtility.getEntryString(StorageKey.CopilotSampleDBEnabled) === "true"
+      : false;
```

Comment:
nit: since `getEntryString()` returns `null` if the key isn't there, you could simplify:
```suggestion
    const copilotSampleDBEnabled = (LocalStorageUtility.getEntryString(StorageKey.CopilotSampleDBEnabled) === "true");
```